src/www/app/outline_server_repository/access_key_serialization.ts

```diff
@@ -31,6 +31,52 @@ export function staticKeyToShadowsocksSessionConfig(staticKey: string): Shadowso
       password: config.password.data,
     };
   } catch (error) {
-    throw new errors.ServerUrlInvalid(error.message || 'Failed to parse static access key.');
+    throw new errors.ServerAccessKeyInvalid(error.message || 'Failed to parse static access key.');
+  }
+}
+
+// fetches information from a dynamic access key and attempts to parse it
+// TODO(daniellacosse): unit tests
+export async function promiseShadowsocksSessionConfigFromDynamicAccessKey(
+  dynamicKey: string
+): Promise<ShadowsocksSessionConfig> {
+  let response, sessionConfig;
+  try {
+    response = await fetch(dynamicKey);
+  } catch (error) {
+    throw new errors.ServerUnreachable(error.message || 'Failed to fetch VPN information from dynamic access key.');
+  }
+
+  const responseBody = (await response.text()).replace(/“|”/g, '"').trim();
```

Comment:
What is this doing? Why do we need this?

Reply:
Remove those weird quotes in the string and removes whitespace from the start and end of it.

Reply:
But those quotes aren't valid anywhere in the JSON or the access key

host: not a valid character here, since it's an IP address or domain:

<img width="439" alt="Screen Shot 2022-10-20 at 15 03 07" src="https://user-images.githubusercontent.com/3759828/197035521-4c836ec9-e9c9-4b59-b20e-7637f06a1980.png">

port: it's a number
password: base64
method: a set of known strings, none of which contain these characters

I don't see how we'd encounter this case you're worried about

Reply:
Are you saying this could be an XSS vector?

Reply:
It could be part of the server name, if we make that a field, or the password.
The password IS NOT base64 encoded. It's a binary string.

Yes, something like XSS